src/build/__main__.py

```diff
@@ -48,10 +48,13 @@ def _format_dep_chain(dep_chain):  # type: (Sequence[str]) -> str
 
 def _build_in_isolated_env(builder, outdir, distributions):
     # type: (ProjectBuilder, str, List[str]) -> None
-    with IsolatedEnvBuilder() as env:
-        builder.python_executable = env.executable
-        env.install(builder.build_dependencies)
-        for distribution in distributions:
+    for distribution in distributions:
+        with IsolatedEnvBuilder() as env:
+            builder.python_executable = env.executable
+            # first install the build dependencies
+            env.install(builder.build_dependencies)
+            # then get the extra required dependencies from the backend
+            env.install(builder.get_dependencies(distribution))
```

Comment:
Should we combined these so we don't have to invoke `pip` twice, which will slow down isolated builds?

Reply:
We first need to install the build dependencies, say setuptools, only after that we can ask the backend, setuptools in this case, which extra dependencies it needs.

Reply:
Ah yes, I didn't think that through.

Reply:
Yeah, PEP-517 spec mandates this to be two separate calls.